server/segments_endpoint_test.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWithValidData(t *testing.T) {
+	// given
+	clearTable()
+	segmentName := "some-new-segment"
+	var responseBody map[string]interface{}
```

Comment:
You could use `Segment` here. Would that make sense?
Then you could also return `[]Segment` from `getSegments()` and you could simply compare the two values you got (of course checking `name` probably makes sense (not sure about checking whether `id` is 1 is useful here, but it may the case))